src/rpc/services/SpectateService.ts

```diff
@@ -0,0 +1,418 @@
+import { ARpcService, MainRpcService } from "@/rpc";
+import { ChannelId, Timestamp } from "@/types/types";
+import { StateSnapshot } from "@/models";
+import Clock from "@/Clock";
+import ATransport from "@/transport/ATransport";
+import {
+    MilestoneProofStruct,
+    StateProofStruct
+} from "@typechain-types/contracts/V1/types/ProofTypes";
+import { DisputeStruct } from "@typechain-types/contracts/V1/types/DisputeTypes";
+import { Codec, Type } from "@/utils";
+import { isEqual } from "lodash";
+
+export interface SnapshotPayload {
+    disputeWindows: DisputeStruct[];
+    stateProof: StateProofStruct;
+    latestForkGenesisSnapshot: StateSnapshot;
```

Comment:
You for sure have to share the `encodedState` that stateProof proves in the end, so they can start replicating the state from that state